sklearnex/_device_offload.py

```diff
@@ -128,23 +128,23 @@ def _transfer_to_host(queue, *data):
 def _get_backend(obj, queue, method_name, *data):
     cpu_device = queue is None or queue.sycl_device.is_cpu
     gpu_device = queue is not None and queue.sycl_device.is_gpu
-    cpu_fallback = False
+    onedal_cpu_supported = obj._onedal_cpu_supported(method_name, *data)
+    onedal_gpu_supported = obj._onedal_gpu_supported(method_name, *data)
 
-    if (cpu_device and obj._onedal_cpu_supported(method_name, *data)) or \
-       (gpu_device and obj._onedal_gpu_supported(method_name, *data)):
-        return 'onedal', queue, cpu_fallback
+    if (cpu_device and onedal_cpu_supported) or \
+       (gpu_device and onedal_gpu_supported):
+        return 'onedal', queue
     if cpu_device:
-        return 'sklearn', None, cpu_fallback
+        return 'sklearn', None
```

Comment:
This change introduce overhead for gpu support check: `_onedal_gpu_supported` was not called before if first condition for cpu is True.

Reply:
Actually, it calls all checks here, but in any way overhead point is correct. Thanks!